### DISCUSSION TITLE
BC lambdas with locally bounded values cannot be hashed?

### DISCUSSION THREAD

**Discussion by marijanarra** [lululxvi/deepxde#776] [Ideas]
Hello,

I wanted to implement different BCs on each edge of a polygon. Doing this requires iteration through the polygon vertices and passing the iterator variable to the BC lambda function as locally bound (like in the piece of code below).

```
for i in polygon_vertices_indices:
      bc = dde.icbc.NeumannBC(geom,
                          lambda x, i=i: bc_val(x, i),
                          lambda x, on_boundary, i=i: on_boundary and on_segment(x, i, i+1))

```
However, if I pass it as locally bound, it seems that `npfunc_range_autocache` returns NoneType.

Is this fixable? I can't think of a proper workaround for my setup :(

Thanks!

**Comment by lululxvi** [comment]
You cannot have the argument `i=i`. The signature must be `lambda x` and `lambda x, on_boundary`. If you don't have a lot of edges, then just write them explicitly. Of course, there are other ways to get around this.
